src/documents/views/VersionPreviewView.js

```diff
@@ -0,0 +1,285 @@
+import React, {useState, useEffect} from 'react';
+import {Helmet} from 'react-helmet';
+import {useQuery} from '@apollo/react-hooks';
+import {GET_FILE_BY_ID} from '../queries';
+import {fileSortedVersions} from '../utilities';
+import {
+  Button,
+  Container,
+  Header,
+  Message,
+  Segment,
+  Popup,
+  Grid,
+  Icon,
+  Label,
+} from 'semantic-ui-react';
+import TimeAgo from 'react-timeago';
+import {longDate} from '../../common/dateUtils';
+import Papa from 'papaparse';
+import * as XLSX from 'xlsx';
+import {LogOnMount} from '@amplitude/react-amplitude';
+import {HotTable} from '@handsontable/react';
+import 'handsontable/dist/handsontable.full.css';
+
+const VersionPreviewView = ({match}) => {
+  const [fullWidth, setFullWidth] = useState(false);
+  const [displayLength, setDisplayLength] = useState(0);
+  const [loading, setLoading] = useState(false);
+  const [data, setData] = useState([]);
+  const [errors, setErrors] = useState([]);
```

Comment:
Instead of `useState` for the data, loading, and errors, you should be able to use the hooks directly to provide that state from Apollo:
```
const {data: fileData, loading: fileLoading, error: fileError} = useQuery(GET_FILE_BY_ID, {
  variables: {kfId: match.params.fileId},
});
```

I think that using the additional state hook may possibly double the number of renders because there's twice as many state variables.